apps/aecore/src/aec_sync.erl

```diff
@@ -188,6 +257,26 @@ terminate(_, _) ->
 code_change(_FromVsn, State, _Extra) ->
     {ok, State}.
 
+insert(#sync_peer{difficulty = Difficulty,
+                  from = From,
+                  to = To,
+                  uri = Uri} = Sync, Pool) ->
+  {NewUri, NewPool} = 
+      case lists:keyfind(Sync#sync_peer.uri, #sync_peer.uri, Pool) of
+          false ->
+             {true, [Sync | Pool]};
+          RSync when RSync#sync_peer.from > From ->
+             {false, lists:keyreplace(Uri, #sync_peer.uri, Pool, 
+                                      RSync#sync_peer{difficulty = 
+                                                        max(Difficulty, RSync#sync_peer.difficulty), 
```

Comment:
I see that `difficulty` and `to` get assigned separately by two max operations. I would expect that one of the two shall drive, and the other shall follow.

----

I see difficulty is used for keeping sync pool sorted, I did not find other usages of it. Not sure whether the sort is actually exploited, it looks not.
The to is actually relevant for understanding when to stop syncing.

I see `difficulty` and `to` come originally from same header - i.e. header advertized by peer as "best hash".

Reply:
On a wider note probably out of scope for this PR, in the ping object the difficulty is taken from top block (i.e. with state) while the best hash from top header. Then I am not sure difficulty is used, neither verified as being part of a header. Also, I am not sure whether "top header hash" is guaranteed to be successor of top block (i.e. with verified state) - as would be required by following header chain for syncing though requiring syncing block at each height (from below).